src/vmm/src/device_manager/persist.rs

```diff
@@ -360,16 +382,26 @@ impl<'a> Persist<'a> for MMIODeviceManager {
                                   slot: &MMIODeviceInfo,
                                   event_manager: &mut EventManager|
          -> Result<(), Self::Error> {
-            dev_manager
-                .slot_sanity_check(slot)
-                .map_err(Error::DeviceManager)?;
-
             let restore_args = MmioTransportConstructorArgs {
                 mem: mem.clone(),
                 device,
             };
             let mmio_transport =
                 MmioTransport::restore(restore_args, state).map_err(|()| Error::MmioTransport)?;
+
+            // At the moment we do not require exact re-allocation of id's via
+            // `dev_manager.irq_allocator.allocate_id()` and at the moment
+            // cannot do this effectively. In the future we may requires this
```

Comment:
Because `IdAllocator` doesn't have an `exact match` policy.

Reply:
let's say something like
```suggestion
            // cannot do this effectively. In the future we may require to preserve `IdAllocator`'s state after snapshot restore
```
or something similar.

Reply:
I've updated the comment [here](https://github.com/firecracker-microvm/firecracker/pull/3001/commits/3a3c441a149035b8a844d4f9038f18519113d136#diff-582bbb144dc948e3c139586a1a98634a5dd227c757db3d0fec10cab78a93b8e8L368-R396) to:
```
// We do not currently require exact re-allocation of IDs via
// `dev_manager.irq_allocator.allocate_id()` and currently cannot do 
// this effectively as `IdAllocator` does not implement an exact 
// match API.
// In the future we may require preserving `IdAllocator`'s state 
// after snapshot restore so as to restore the exact interrupt IDs 
// from the original device's state for implementing hot-plug.
// For now this is why we do not restore the state of the
// `IdAllocator` under `dev_manager`.
```
Do you think this covers your points?